File: Code/22_Temporizacao/Ex6.c
#include <stdio.h>
#include <time.h>
#include <unistd.h>

#define N 150
#define DELAY_US 10000 
int main(void)
{
	unsigned int i;
	struct timespec t[2];
	double media;

	clock_gettime(CLOCK_MONOTONIC, &t[0]);
	for(i=1; i<=N; i++)
		usleep(DELAY_US);
	clock_gettime(CLOCK_MONOTONIC, &t[1]);
	media = difftime(t[1].tv_sec,
		t[0].tv_sec)*1.0E9;
	media += (float)(t[1].tv_nsec -
			t[0].tv_nsec);
	media /= (float)N;
	printf("Media de tempo de usleep(%d)\n"
		"   N =  %d, %2.1f us\n",
		DELAY_US, N, media/1000.0);
}

File: 4_Dicas/4.4_Temporização/Python/Ex6.py
import time

N = 150
DELAY_US = 10000
DELAY_S  = DELAY_US*1e-6

CPU_time = [0,0]
CPU_time[0] = time.clock_gettime(time.CLOCK_MONOTONIC)
for i in range(N):
	time.sleep(DELAY_S)
CPU_time[1] = time.clock_gettime(time.CLOCK_MONOTONIC)
media = (CPU_time[1]-CPU_time[0])/N

print("Media de tempo de time.sleep(%f)\n"
	"   N =  %d, %2.1f us" % (DELAY_S, N, media*1.0e6))